src/transactions/simulation/TxSimGenerateBucketsWork.cpp

```diff
@@ -152,14 +149,9 @@ TxSimGenerateBucketsWork::onRun()
         auto bucketHash = mIsCurr ? currentLevel.curr : currentLevel.snap;
         auto bucket =
             mApp.getBucketManager().getBucketByHash(hexToBin256(bucketHash));
-        Hash emptyHash;
-        if (bucket->getHash() != emptyHash)
-        {
-            CLOG(INFO, "History")
-                << "Simulating " << (mIsCurr ? "curr" : "snap")
-                << " bucketlist level: " << mLevel;
-            startBucketGeneration(bucket);
-        }
+        CLOG(INFO, "History") << "Simulating " << (mIsCurr ? "curr" : "snap")
```

Comment:
Can you explain which issue is getting fixed?

The other change is a straight rollback of 9ac04eb5fca4d9bd0fd3b3135e8d21d115989d1c - but what is the actual problem here?

From what I see
* the new behavior is invoking `startBucketGeneration` with an empty bucket to populate `mIntermediateBuckets` with a bunch of empty buckets that end up getting merged into an empty bucket - so we don't special case "empty bucket" anymore
* but the old behavior was using `mIntermediateBuckets.empty()` as a proxy for "empty bucket"

I can't tell why those two behaviors end up producing different results.

Reply:
The problem here is not related to empty buckets, it's just that by trying to accommodate empty buckets, I broke generation of normal buckets. The issue with https://github.com/stellar/stellar-core/commit/9ac04eb5fca4d9bd0fd3b3135e8d21d115989d1c is that on the first `onRun` it'd skip the first bucket -- `curr` of level 0. So there were two cases where `mIntermediateBuckets.empty()`: at the very beginning (we haven't started merging), and when we encounter empty buckets (we just skip that bucket). 

I don't think handling empty buckets really matters here: transaction replay requires protocol 12 or higher (so no buckets with empty hashes), and performance-wise, it's the buckets with contents that make a difference.

Reply:
ah yes the first iteration was different